modules/audiencelogyBidAdapter.js

```diff
@@ -0,0 +1,161 @@
+      requests.push(request);
+      // Set nid value
+      nid = bid.params.nid;
+    });
+    // Return the array of bid requests
+    return {
+      method: 'POST',
+      url: `${ENDPOINT_URL}/${nid}`,
+      data: JSON.stringify(requests),
+      options: {
+        contentType: 'application/json',
+      }
+    };
+  },
+  // Unpack the response from the server into a list of bids.
+  interpretResponse: function (bidResponse, bidRequest) {
+    let resp = [];
+    if (bidResponse && bidResponse.body) {
+      try {
+        let bids = bidResponse.body.seatbid && bidResponse.body.seatbid[0] ? bidResponse.body.seatbid[0].bid : [];
+        if (bids) {
+          bids.forEach(bidObj => {
+            let newBid = buildResponse(bidObj);
+            newBid.mediaType = BANNER;
+            resp.push(newBid);
+          });
+        }
+      } catch (err) {
+        logError(err);
+      }
+    }
+    return resp;
+  }
+}
+// Function to fetch bid_floor
+const fetchFloor = (bid) => {
+  if (bid.params && bid.params.bid_floor) {
+    return bid.params.bid_floor;
+  } else {
+    return 0;
+  }
+}
+// Function to fetch banner details
+const fetchBanner = (bid) => {
+  if (deepAccess(bid, 'mediaTypes.banner')) {
+    // Fetch width and height from MediaTypes object, if not provided in bid params
+    if (deepAccess(bid, 'mediaTypes.banner.sizes') && !bid.params.height && !bid.params.width) {
+      let sizes = deepAccess(bid, 'mediaTypes.banner.sizes');
+      if (isArray(sizes) && sizes.length > 0) {
+        return {
+          h: sizes[0][1],
+          w: sizes[0][0]
+        };
+      }
+    } else {
+      return {
+        h: bid.params.height,
+        w: bid.params.width
+      };
+    }
+  }
+}
+// Function to fetch site details
+const fetchSite = (bidderRequest) => {
+  let site = {};
+  if (bidderRequest && bidderRequest.refererInfo && bidderRequest.refererInfo.page) {
+    site.name = bidderRequest.refererInfo.domain;
+  } else {
+    site.name = '';
+  }
+  return site;
+}
+// Function to create response
+const buildResponse = (bid) => {
+  let response = {};
+  response.requestId = bid && bid.impid ? bid.impid : undefined;
+  response.width = bid && bid.w ? bid.w : 0;
+  response.height = bid && bid.h ? bid.h : 0;
+  response.cpm = bid && bid.price ? bid.price : 0.0;
+  response.ad = bid && bid.adm ? bid.adm : '';
+  response.creativeId = bid && bid.crid ? bid.crid : undefined;
+  response.meta = {
+    advertiserDomains: bid && bid.adomain ? bid.adomain : []
+  };
+  response.netRevenue = false;
+  response.currency = bid && bid.cur ? bid.cur : 'USD';
+  response.dealId = bid && bid.dealId ? bid.dealId : undefined;
+  response.ttl = 300;
+  return response;
+}
+// Function to build the user object
+const setUser = (bid) => {
+  let user = {};
+  if (bid && bid.params) {
+    user.buyeruid = localStorage.getItem('adx_profile_guid') ? localStorage.getItem('adx_profile_guid') : '';
```

Comment:
You have to import the storage manager to do this, publishers manage bidder access to storage depending on consent or their preferences